packages/react-core/src/components/ApplicationLauncher/examples/ApplicationLauncherBasic.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { ApplicationLauncher, ApplicationLauncherItem } from '@patternfly/react-core';
+
+const appLauncherItems: React.ReactElement[] = [
+  <ApplicationLauncherItem key="application_1a" href="#">
+    Application 1 (anchor link)
+  </ApplicationLauncherItem>,
+  <ApplicationLauncherItem key="application_2a" component="button" onClick={() => alert('Clicked item 2')}>
+    Application 2 (button with onClick)
+  </ApplicationLauncherItem>,
+  <ApplicationLauncherItem key="disabled_application_4a" isDisabled>
+    Unavailable application
+  </ApplicationLauncherItem>
+];
+
+export const ApplicationLauncherBasic: React.FunctionComponent = () => {
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+  const onToggle = React.useCallback((isOpen: boolean) => setIsOpen(isOpen), []);
+  const onSelect = React.useCallback((_event: any) => setIsOpen(prevIsOpen => !prevIsOpen), []);
```

Comment:
I'm curious, what is the reasoning for the `setIsOpen(prevIsOpen => !prevIsOpen)` notation vs just negating the current state variable directly? Is it a requirement of using `useCallback` or just a style preference?

Reply:
It is not a requirement. You can do either, both work fine, but since the function is memoized, we would have to add the isOpen state variable as a dependency, so utilizing the ability to access previous state from the setter is a little less code and a little cleaner.